src/WebJobs.Extensions.DurableTask.Analyzers/Analyzers/ActivityFunction/FunctionAnalyzer.cs

```diff
@@ -128,27 +142,92 @@ private bool TryGetFunctionNameFromActivityInvocation(InvocationExpressionSyntax
             return false;
         }
 
-        private bool TryGetInputNodeFromCallActivityInvocation(InvocationExpressionSyntax invocationExpression, out SyntaxNode inputNode)
+        private static bool TryGetInputNodeFromCallActivityInvocation(SemanticModel semanticModel, InvocationExpressionSyntax invocationExpression, out SyntaxNode inputNode)
+        {
+            // If CallActivity is a custom method in user code
```

Comment:
Sure, this is meant to cover custom overloads of CallActivity* methods. We can only get the declared syntax node if the method is defined in the compilation were analyzing.

See my description at "Updated FunctionAnalyzer to avoid flagging input incorrectly on custom CallActivity methods," for an example on the linked issue.

Reply:
It's probably best to describe this as a custom CallActivity _extension_ method. The customer-provided repro in [this issue](https://github.com/Azure/azure-functions-durable-extension/issues/1562) shows a good example, and it looks like Amanda is using that exact implementation in the test case.